.github/workflows/release.yml

```diff
@@ -21,38 +21,46 @@ jobs:
 
     steps:
       # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
           repository: wingtk/gvsbuild
           fetch-depth: 0
 
       - name: Set up Python
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v3
         with:
           python-version: ${{ matrix.python }}
           architecture: ${{ matrix.arch }}
 
       - name: Install Build Deps
         run: python -m pip install wheel
 
+      - name: Install Poetry
+        run: python -m pip install poetry==1.1.13
+      - name: Configure Poetry
+        run: poetry config virtualenvs.in-project true
+      - name: Install Python Dependencies
+        run: poetry install
+
       - name: Move git binary
         # Temporarily move preinstalled git to prevent errors related to cygwin.
         run: Rename-Item "C:\Program Files\Git\usr\bin" notbin
 
+      - name: Build gvsbuild
+        run: pip install .
+
       - name: Build
         run: >
-          python build.py build
+          gvsbuild build
           --platform=${{ matrix.arch }}
           --vs-ver=${{ matrix.vstudio }}
           --same-python
           --enable-gi
           --py-wheel
-          --skip gtksourceview,emeus,clutter
+          --patches-root-dir=${{ github.workspace }}\patches
+          --skip gtksourceview4,emeus,clutter
```

Comment:
For the CI run we only need to have the `gvsbuild` command available which is either via `poetry run ...` or `pip install .`. However we should not be mixing install process as this is extra unneeded steps.

The main reason for using poetry would be a consistent build for the CI since it would use the poetry lock file and fixed version of poetry so could you update this to match the gvsbuild CI script

Reply:
This version of the CI file only uses pip install . and gvsbuild, NOT poetry run gvsbuild.... So its not mixing? (if I understand correctly) however I can test changing it to how it is in their CI if you would like. Again I was going by their readme which they should update if thats not the proper way to build or best way to build but thats them not us.